### ISSUE TITLE
py 1.8.2 breaks collect-only package printing on windows

### ISSUE THREAD

**Issue by krzysztofmurawski** [pytest-dev/pytest#7477]
When having a tests directory that contains subfolders with each containing some tests:
e.g. 

```
tests
  folder_1
  folder_2
    subfolder_1
      test_module.py
```

collect only always only reports Package as tests
e.g.:
```
<Package tests>
  <Module test_module.py>
    <Function test_someting[parameter]>
```
 
1.8.1 on windows and 1.8.2 on Darwin / linux would do this:
e.g.:
```
<Package tests\folder_2\subfolder_1>
  <Module test_module.py>
    <Function test_someting[parameter]>
```

**Comment by nicoddemus** [comment]
I can reproduce this using the zip file, thanks. This is what I get:

```
λ pytest --collect-only
======================================= test session starts =======================================
platform win32 -- Python 3.7.6, pytest-6.0.0rc2.dev7+g906d8496c.d20200710, py-1.9.0, pluggy-0.13.1
rootdir: d:\projects\pytest\.tmp\py_issue, configfile: pytest.ini
plugins: hypothesis-5.14.0, forked-1.1.3, xdist-1.33.0
collected 1 item

<Package subfolder_1>
  <Module test_module.py>
    <Function test_something[parameter]>

====================================== no tests ran in 0.04s ======================================
```

This is what we should be seeing instead:

```
λ pytest --collect-only
======================================= test session starts =======================================
platform win32 -- Python 3.7.6, pytest-6.0.0rc2.dev7+g906d8496c.d20200710, py-1.9.0, pluggy-0.13.1
rootdir: d:\projects\pytest\.tmp\py_issue, configfile: pytest.ini
plugins: hypothesis-5.14.0, forked-1.1.3, xdist-1.33.0
collected 1 item

<Package tests>
  <Package subfolder_1>
    <Module test_module.py>
      <Function test_something[parameter]>

====================================== no tests ran in 0.04s ======================================
```

It is somehow not collecting `tests` even thought it contains an `__init__.py` file:

```
│   pytest.ini
│   __init__.py
│
├───tests
│   │   __init__.py
│   │
│   ├───folder_1
│   │       __init__.py
│   │
│   └───folder_2
│       │   __init__.py
│       │
│       └───subfolder_1
│               test_module.py
│               __init__.py
```

**Comment by nicoddemus** [comment]
Actually I get the same behavior on `1.8.1`, which leads me to believe the problem is not clear (at least to me).

This is what I get in `5.4.3`:

```
λ pytest .tmp\py_issue\tests --collect-only
======================== test session starts ========================
platform win32 -- Python 3.7.6, pytest-5.4.3, py-1.8.0, pluggy-0.13.1
rootdir: d:\projects\pytest\.tmp\py_issue, inifile: pytest.ini
plugins: hypothesis-5.14.0, forked-1.1.3, xdist-1.33.0
collected 1 item
<Package D:\projects\pytest\.tmp\py_issue\tests\folder_2\subfolder_1>
  <Module test_module.py>
    <Function test_something[parameter]>

======================= no tests ran in 0.04s =======================
```

And here in `6.0.0`/`master`/`py-1.8.0`:

```
λ pytest .tmp\py_issue\tests --collect-only
======================== test session starts ========================
platform win32 -- Python 3.7.6, pytest-5.4.3, py-1.8.0, pluggy-0.13.1
rootdir: d:\projects\pytest\.tmp\py_issue, configfile: pytest.ini
plugins: hypothesis-5.14.0, forked-1.1.3, xdist-1.33.0
collected 1 item

<Package subfolder_1>
  <Module test_module.py>
    <Function test_something[parameter]>

======================= no tests ran in 0.03s =======================
```

And I get the exact same output in `6.0.0`/`master` and `py-1.8.1`, `py-1.8.2`, and `py-1.9.0`.

(My comment in https://github.com/pytest-dev/pytest/issues/7477#issuecomment-656891323 was incorrect).

Notice that the same packages are still being collected, only their `name` has changed, from `"D:\projects\pytest\.tmp\py_issue\tests\folder_2\subfolder_1"` to `"subfolder_1"`. This was done intentionally in https://github.com/pytest-dev/pytest/pull/7046, for consistency with how all nodes are named.

So I'm not sure, it seems everything is working as intended.

@krzysztofmurawski can you elaborate on what problem you are having?

**Comment by mdmintz** [comment]
@nicoddemus You might also want to ask the Microsoft VS Code people who ran into the issue here: https://github.com/microsoft/vscode-python/issues/12372

**Comment by nicoddemus** [comment]
Managed to understand the problem, but not sure how to proceed: https://github.com/microsoft/vscode-python/issues/12372#issuecomment-659392819

**Comment by bluetech** [comment]
> So whatever the problem was it will be fixed in pytest-6.0.0.

I did not think of this possibility, that explains why no matter what I tried I couldn't reproduce it.

It will be interesting to bisect this, not sure I will get to it myself though (maybe).

**Comment by nicoddemus** [comment]
Closing for now as this has been fixed in `master` already. Let's see if this becomes a more serious problems to warrant a 5.4.4 release.
